autoprof/fit/langevin.py

```diff
@@ -107,37 +134,56 @@ def accept(self, log_alpha):
         """
         return torch.log(torch.rand(log_alpha.shape, device=self.model.device)) < log_alpha
 
-    def delta_logp(self, current_state, proposed_state):
+    def delta_logp(self, current_state: torch.Tensor, proposed_state: torch.Tensor)-> torch.Tensor:
         """
         Computes the log probability difference between the proposed state x' and the current state x
             log p(x') - log p(x)
         """
         v = proposed_state - current_state
         T = self.delta_logp_steps + 1
         D = current_state.shape
+        return simps(self.integrand, 0., 1., self.delta_logp_steps, device=self.model.device)
 
-        def integrand(t):
+    def integrand(self, t: torch.Tensor)-> torch.Tensor:
```

Comment:
The `integrand` function is intentionally defined inside the the `delta_logp` function for simplicity. I don't think this needs to change

Reply:
Okay I'll revert it back...